code/obj/item/mob_parts/artifact_parts.dm

```diff
@@ -0,0 +1,540 @@
+		handlistPart = "leg-eldritch-L-attached"
+
+	right
+		name = "eldritch right leg"
+		slot = "r_leg"
+		side = "right"
+		step_image_state = "footprintsR"
+		icon_state = "leg-eldritch-R"
+		handlistPart = "leg-eldritch-R-attached"
+
+/obj/item/parts/artifact_parts/arm/martian
+	artifact_type = "martian"
+	update_with_clothing = TRUE
+
+	on_attach()
+		if (!..())
+			return
+		var/mob/living/carbon/human/H = holder
+		if (istype(H.limbs.get_limb("l_arm"), /obj/item/parts/artifact_parts/arm/martian/left) && istype(H.limbs.get_limb("r_arm"), /obj/item/parts/artifact_parts/arm/martian/right))
+			src.holder.addAbility(/datum/targetable/artifact_limb_ability/martian_pull)
+
+	on_remove()
+		if (!..())
+			return
+		src.holder.removeAbility(/datum/targetable/artifact_limb_ability/martian_pull)
+
+	left
+		name = "martian left arm"
+		slot = "l_arm"
+		side = "left"
+		icon_state = "arm-martian-L"
+		handlistPart = "arm-martian-L-attached"
+
+	right
+		name = "martian right arm"
+		slot = "r_arm"
+		side = "right"
+		icon_state = "arm-martian-R"
+		handlistPart = "arm-martian-R-attached"
+
+/obj/item/parts/artifact_parts/leg/martian
+	artifact_type = "martian"
+
+	left
+		name = "martian left leg"
+		slot = "l_leg"
+		side = "left"
+		step_image_state = "footprintsL"
+		icon_state = "leg-martian-L"
+		partlistPart = "leg-martian-L-attached"
+		movement_modifier = /datum/movement_modifier/martian_legs/left
+
+	right
+		name = "martian right leg"
+		slot = "r_leg"
+		side = "right"
+		step_image_state = "footprintsR"
+		icon_state = "leg-martian-R"
+		partlistPart = "leg-martian-R-attached"
+		movement_modifier = /datum/movement_modifier/martian_legs/right
+
+/obj/item/parts/artifact_parts/arm/precursor
+	artifact_type = "precursor"
+	update_with_clothing = TRUE
+
+	on_attach()
+		if (!..())
+			return
+		var/mob/living/carbon/human/H = holder
+		if (istype(H.limbs.get_limb("l_arm"), /obj/item/parts/artifact_parts/arm/precursor/left) && istype(H.limbs.get_limb("r_arm"), /obj/item/parts/artifact_parts/arm/precursor/right))
+			src.holder.addAbility(/datum/targetable/artifact_limb_ability/precursor_heal)
+
+	on_remove()
+		if (!..())
+			return
+		src.holder.removeAbility(/datum/targetable/artifact_limb_ability/precursor_heal)
+
+	left
+		name = "precursor left arm"
+		slot = "l_arm"
+		side = "left"
+		icon_state = "arm-precursor-L"
+		handlistPart = "arm-precursor-L-attached"
+
+	right
+		name = "precursor right arm"
+		slot = "r_arm"
+		side = "right"
+		icon_state = "arm-precursor-R"
+		handlistPart = "arm-precursor-R-attached"
+
+/obj/item/parts/artifact_parts/leg/precursor
+	artifact_type = "precursor"
+
+	on_attach()
+		if (!..())
+			return
+		if (src.side == "left")
+			RegisterSignal(src.holder, COMSIG_MOVABLE_MOVED, .proc/precursor_move_L)
+		else
+			RegisterSignal(src.holder, COMSIG_MOVABLE_MOVED, .proc/precursor_move_R)
+
+	on_remove()
+		if (!..())
+			return
+		UnregisterSignal(src.holder, COMSIG_MOVABLE_MOVED)
```

Comment:
If only one is removed does that stop it?

Reply:
Nope